contracts/governance/Vesting/fouryear/FourYearVesting.sol

```diff
@@ -0,0 +1,64 @@
+pragma solidity ^0.5.17;
+pragma experimental ABIEncoderV2;
+
+import "../../../openzeppelin/Ownable.sol";
+import "../../../interfaces/IERC20.sol";
+import "../../Staking/Staking.sol";
+import "../../IFeeSharingProxy.sol";
+import "../../ApprovalReceiver.sol";
+import "./FourYearVestingStorage.sol";
+import "../../../proxy/Proxy.sol";
+
+/**
+ * @title Four Year Vesting Contract.
+ *
+ * @notice A four year vesting contract.
+ *
+ * @dev Vesting contracts shouldn't be upgradable,
+ * use Proxy instead of UpgradableProxy.
+ * */
+contract FourYearVesting is FourYearVestingStorage, Proxy {
+	/**
+	 * @notice Setup the vesting schedule.
+	 * @param _logic The address of logic contract.
+	 * @param _SOV The SOV token address.
+	 * @param _tokenOwner The owner of the tokens.
+	 * @param _cliff The time interval to the first withdraw in seconds.
+	 * @param _duration The total duration in seconds.
+	 * @param _feeSharingProxy Fee sharing proxy address.
+	 * */
+	constructor(
+		address _logic,
+		address _SOV,
```

Comment:
we tend to pass SOV address as a parameter all over the contracts but it is actually a constant. why not just hardcode it as a constant?

Reply:
SOV address can be different based on the network I guess - as for other addresses like Staking, FeeSharing etc.